CPS-????/README.md

```diff
@@ -0,0 +1,242 @@
+
+Being that the atomic swap script checks all the inputs and outputs in the transaction to determine validity, below is the transaction fee estimation with the current limitation:
+
+``` Txt
+tx fee = 0.3 ADA * #swap_ref_script_executions + 0.02 ADA * ( #input_utxos + #output_utxos)
+``` 
+
+As the equation shows, when the swap script must be exected once per UTxO, the fee can grow quite large very quickly. Meanwhile, if the redundant exectutions were not there, the fee would grow very slowly.
+
+## The Trick To Minimize This Limitation
+The trick involves taking advantage of the fact that staking scripts do not suffer from the same limitation as spending scripts.
+
+> Given a [spending] script **s**, you can turn it into a [spending] script **s'** that checks that 0 ADA is withdrawn from a [staking] script **s''** with the same logic as **s**. This is a transformation that is valid for any [spending] script.    -- @L-as
+
+
+In a nut shell, you use a staking script **AS** a spending script. Here is how the above transaction example would look with this trick:
+
+``` Bash
+cardano-cli transaction build \
+  --tx-in c1d01ea50fd233f9fbaef3a295ba607a72c736e58c9c9df588abf4e5009ad4fe#0 \
+  --tx-in 622034715b64318e9e2176b7ad9bb22c3432f360293e9258729ce23c1999b9d8#2 \
+  --spending-tx-in-reference 622034715b64318e9e2176b7ad9bb22c3432f360293e9258729ce23c1999b9d8#0 \
+  --spending-plutus-script-v2 \
+  --spending-reference-tx-in-inline-datum-present \
+  --spending-reference-tx-in-redeemer-file $swapRedeemerFile \
+  --tx-in 766555130db8ff7b50fc548cbff3caa0d0557ce5af804da3b993cd090f1a8c3a#1 \
+  --spending-tx-in-reference 622034715b64318e9e2176b7ad9bb22c3432f360293e9258729ce23c1999b9d8#0 \
+  --spending-plutus-script-v2 \
+  --spending-reference-tx-in-inline-datum-present \
+  --spending-reference-tx-in-redeemer-file $swapRedeemerFile \
+  --tx-out "$(cat ../assets/wallets/01.addr) 2000000 lovelace + 300 c0f8644a01a6bf5db02f4afe30d604975e63dd274f1098a1738e561d.54657374546f6b656e0a + 0 c0f8644a01a6bf5db02f4afe30d604975e63dd274f1098a1738e561d.4f74686572546f6b656e0a" \
+  --tx-out "$(cat ${swapScriptAddrFile}) + 4000000 lovelace + 0 c0f8644a01a6bf5db02f4afe30d604975e63dd274f1098a1738e561d.54657374546f6b656e0a + 250 c0f8644a01a6bf5db02f4afe30d604975e63dd274f1098a1738e561d.4f74686572546f6b656e0a" \
+  --tx-out-inline-datum-file $swapDatumFile \
+  --withdrawal "$(cat ${stakingScriptAddrFile})+0" \
+  --withdrawal-script-file $stakingScriptFile \
+  --withdrawal-redeemer-file $swapRedeemerFile \
+  --tx-in-collateral bc54229f0755611ba14a2679774a7c7d394b0a476e59c609035e06244e1572bb#0 \
+  --change-address $(cat ../assets/wallets/01.addr) \
+  --protocol-params-file "${tmpDir}protocol.json" \
+  --testnet-magic 1 \
+  --out-file "${tmpDir}tx.body"
+```
```

Comment:
Same remark as above. The cardano-cli command is rather opaque and doesn't highlight well what the workaround is about. I'd suggest to reformulate as prose or a diagram.